Dockerfile-windows.template

```diff
@@ -0,0 +1,51 @@
+FROM mcr.microsoft.com/windows/servercore:version as installer
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+# PATH isn't actually set in the Docker image, so we have to set it from within the container
+RUN $newPath = ('C:\Program Files (x86)\GnuPG\bin;{0}' -f $env:PATH); \
+	Write-Host ('Updating PATH: {0}' -f $newPath); \
+	[Environment]::SetEnvironmentVariable('PATH', $newPath, [EnvironmentVariableTarget]::Machine)
+# doing this first to share cache across versions more aggressively
+
+ENV NODE_VERSION 0.0.0
+ENV NODE_CHECKSUM CHECKSUM_x64
+
+ENV GPG_VERSION 2.4.5_20240307
+
+RUN Invoke-WebRequest $('https://www.gnupg.org/ftp/gcrypt/binary/gnupg-w32-{0}.exe' -f $env:GPG_VERSION) -OutFile 'gpg-installer.exe'; \
+	Start-Process -FilePath 'gpg-installer.exe' -ArgumentList '/S' -Wait; \
```

Comment:
I could check the integrity through `gpg --verify gnupg-2.2.44.exe.sig gnupg-2.2.44.exe`, for example. But I don't think it makes sense, as if I download the program through which I then verify that program itself, it means that if the program gets compromised, it doesn't matter. Though I could probably create a static checksum of a program and verify it through that as I don't think that program will be updated on daily basis. What do you think?

Reply:
Yeah! That's definitely better! We usually use static checksums in Docker Official Images.

Small knit, is it possible to use a longer algorithm like sha256 instead of sha1?

Reply:
That shouldn't be a problem, although I cannot find any SHA256 in the official email announcements that they suggest using as the source of truth. However, they do provide SHA256 values on their website, so we can take them from there. [https://gnupg.org/download/integrity_check.html](https://gnupg.org/download/integrity_check.html)

Reply:
Updated to use sha256